bin/somalier_wrapper.py

```diff
@@ -0,0 +1,243 @@
+	df = pd.read_csv(file, usecols=["sample_id", identifier])
+
+	# Check that the given patient identifier column is present in the sample info csv
+	if identifier not in df.columns:
+		raise ValueError(
+			f"The identifier '{identifier}' is not present in the csv file '{file}'."
+		)
+
+	# Group by patient identifier and join sample IDs
+	grouped = (
+		df.groupby(identifier)["sample_id"].apply(lambda x: ",".join(x)).reset_index()
+	)
+
+	# Write to file without quotes
+	output_file = "somalier_output/expected_relationships.csv"
+	with open(output_file, "w") as f:
+		for item in grouped["sample_id"]:
+			print(item)
+			f.write(f"{item}\n")
+
+	# Get list of .somalier files in the specified directory
+	# If no files present return error
+	somalier_files = glob.glob(
+		"somalier_output/temp/*.somalier"
+	)  ## <- update so that only looks at specified samples
+	if not somalier_files:
+		logging.warning(
+			"No .somalier files found in 'somalier_output/temp/' directory."
+		)
+		return
+
+	# Join list into a single string
+	somalier_files_str = " ".join(somalier_files)
+
+	# Construct command for running somalier relate
+	cmd = (
+		f"somalier relate "
+		f"{somalier_files_str} "
+		f"--output-prefix somalier_output/{prefix} "
+		f"--groups somalier_output/expected_relationships.csv"
+	)
+
+	# Run command
+	try:
+		subprocess.run(cmd, shell=True, check=True)
+		logging.info(
+			f"Relatedness analysis completed with groups file: somalier_output/expected_relationships.csv"
+		)
+	except subprocess.CalledProcessError as e:
+		logging.error("Error occurred during somalier relate")
+		logging.error(str(e))
+
+
+############################################################################################################
+# Estimate sample ancestries
+############################################################################################################
+
+def somalier_ancestry(path_1K, ancestry_labels, prefix):
+
+	# Construct command - uses wildcards rather than lists of files as runs into OSError due to exceeding
+	# max argument length
+	cmd = (
+    f"somalier ancestry --labels {ancestry_labels} "
+    f"{path_1K}/*.somalier ++ somalier_output/temp/*.somalier "
+    f"--output-prefix somalier_output/{prefix} "
+	)
+
+	# Run the command
+	try:
+		subprocess.run(cmd, shell=True, check=True)
+		logging.info(f"Ancestry prediction completed")
+	except subprocess.CalledProcessError as e:
+		logging.error("Error occurred during somalier ancestry prediction")
+		logging.error(str(e))
+
+
+############################################################################################################
+# Main function
+############################################################################################################
+
+def main():
+	# Set log format
+	logging.basicConfig(
+		level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+	)
+
+	# Gather command line arguments
+	args = parse_arguments()
```

Comment:
Do we want to do an upfront check that if you're running with relatedness turned on that you've supplied all three arguments? Would save running something long for it to crash part way